src/utils.ts

```diff
@@ -251,12 +251,29 @@ export function decorateWithReadConcern(
  * @param command - the command on which to apply the explain
  * @param options - the options containing the explain verbosity
  */
-export function decorateWithExplain(command: Document, explain: Explain): Document {
-  if (command.explain) {
+export function decorateWithExplain(
+  command: Document,
+  explain: Explain
+): {
+  explain: Document;
+  verbosity: ExplainVerbosity;
+  maxTimeMS?: number;
+} {
+  type ExplainCommand = ReturnType<typeof decorateWithExplain>;
```

Comment:
This type changes to whatever I make this function return, the self referencing means I don't really get any warning if I do something unexpected with the return type.

Reply:
That's not the point of this type.  It's only here to satisfy the return case when we already have a key of `explain` in the document.

Two other options, if you'd prefer:
- remove this and the type assertion, revert the logic change and just type cast this as-any
- remove this logic entirely.  it's unused (we never decorate a command twice).  I chose to leave it in because it seemed like a nice safeguard in case that changes in the future.

Reply:
It depends on what is important to maintain 🤔 maybe casting makes sense in order to not change the existing functionality. But if it is true that we never can rewrap, then maybe this isn't necessary. 

Maybe this was added if explain gets retried?